FINALLY modularizes Polychromic Clothes AND modernizes all polychromic code!

Michiyamenotehifunana opened the pull request:
🆑 Toriate
add: Added Polychromic Athletic Shorts
imageadd: added greyscaled athletic shorts
tweak: optimized polycloth code, modularized it more
tweak: all polychromic color inputs now use the previously selected color as the default argument, as per #5090 
/🆑


![dreamseeker_2017-12-15_11-47-53](https://user-images.githubusercontent.com/31995558/34026460-319867f4-e191-11e7-8602-a17e9bd729ed.png)

This PR optimizes polychromic clothing code and properly documents it to allow anyone to easily add their own RGB clothes. In addition, polychromic athletic shorts have been added.

**NOW WITH NO MORE CONFLICTS, 100% BAMBOOZLE FREE!**

**EDIT:** All known polychromic color inputs have been updated to be in line with #5090

Michiyamenotehifunana committed:
FINALLY AFTER 2 MONTHS

WOOOOOOOOOOOOOOO

deathride58 reviewed (changes requested):
>claims "No conflicts" in PR body
 >Has .dme conflict

This is false advertisement, I want a refund!

Michiyamenotehifunana committed:
Merge branch 'master' into Modularized-Polychrome-The-Third

Michiyamenotehifunana committed:
Updates all polychromic code

This makes the default argument of all polychromic color input menus the last chosen color.
